addons/website_sale/views/product_views.xml

```diff
@@ -98,6 +98,42 @@
         </field>
     </record>
 
+    <record model="ir.ui.view" id="product_template_only_website_form_view">
+        <field name="name">product.template.product.only.website.form</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_only_form_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='uom_id']" position="after">
+                <label for="base_unit_count" groups="website_sale.group_show_uom_price" attrs="{'invisible': [('product_variant_count', '>', 1)]}"/>
```

Comment:
Not a big fan.. weird when you switch from one product.template to another to see this field disapear (or not appear when you search for it on a product.template)

I think a better UX/UI would be to at least show somehow a message "As you have multiple variants, base unit price has to be configured on variant directly", maybe as a tooltip on top of a readonly element?

Not sure.. What do you think?